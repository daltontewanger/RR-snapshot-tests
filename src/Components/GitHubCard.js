import React from "react";
import Card from "react-bootstrap/Card";

function GitHubCard({ userData }) {
  return (
    <div className="d-flex justify-content-center align-items-center">
      <Card style={{ width: "20rem" }}>
        <Card.Img variant="top" src={userData.avatar_url} alt="profile" />
        <Card.Body>
          <Card.Title>{userData.login}</Card.Title>
          <Card.Text>
            <a href={userData.html_url} target="blank">
              Profile
            </a>
            {userData.bio && <p>Bio: {userData.bio}</p>}
          </Card.Text>
        </Card.Body>
      </Card>
    </div>
  );
}

export default GitHubCard;
