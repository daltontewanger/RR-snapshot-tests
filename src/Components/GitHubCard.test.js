import React from 'react';
import renderer from 'react-test-renderer';
import GitHubCard from './GitHubCard';

describe('GitHubCard', () => {
  it('renders a snapshot', () => {
    const userData = {
      avatar_url: 'https://example.com/avatar.png',
      login: 'testuser',
      html_url: 'https://github.com/testuser',
      bio: 'This is a test bio'
    };
    const tree = renderer.create(<GitHubCard userData={userData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
