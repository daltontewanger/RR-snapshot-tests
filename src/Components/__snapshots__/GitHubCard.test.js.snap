// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitHubCard renders a snapshot 1`] = `
<div
  className="d-flex justify-content-center align-items-center"
>
  <div
    className="card"
    style={
      Object {
        "width": "20rem",
      }
    }
  >
    <img
      alt="profile"
      className="card-img-top"
      src="https://example.com/avatar.png"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        testuser
      </div>
      <p
        className="card-text"
      >
        <a
          href="https://github.com/testuser"
          target="blank"
        >
          Profile
        </a>
        <p>
          Bio: 
          This is a test bio
        </p>
      </p>
    </div>
  </div>
</div>
`;
